src/expirable_map.rs

```diff
@@ -0,0 +1,144 @@
+//! Provides a map that associates values with keys and supports expiring entries.
+//!
+//! Designed for performance-oriented use-cases utilizing `FxHashMap` under the hood,
+//! and is not suitable for cryptographic purposes.
+
+use rustc_hash::FxHashMap;
+use std::{
+    hash::Hash,
+    time::{Duration, Instant},
+};
+
+#[derive(Clone, Debug)]
+pub struct ExpirableEntry<V> {
+    pub(crate) value: V,
+    pub(crate) expires_at: Instant,
+}
+
+impl<V> ExpirableEntry<V> {
+    #[allow(dead_code)]
+    pub fn get_element(&self) -> &V {
+        &self.value
+    }
+
+    #[allow(dead_code)]
+    pub fn update_expiration(&mut self, expires_at: Instant) {
+        self.expires_at = expires_at
+    }
+}
+
+impl<K: Eq + Hash, V> Default for ExpirableMap<K, V> {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+/// A map that allows associating values with keys and expiring entries.
+/// It is important to note that this implementation does not automatically
+/// remove any entries; it is the caller's responsibility to invoke `clear_expired_entries`
+/// at specified intervals.
+///
+/// WARNING: This is designed for performance-oriented use-cases utilizing `FxHashMap`
+/// under the hood and is not suitable for cryptographic purposes.
+#[derive(Clone, Debug)]
+pub struct ExpirableMap<K: Eq + Hash, V>(FxHashMap<K, ExpirableEntry<V>>);
```

Comment:
`ExpirableMap` object and its methods are duplicated in KomodoDeFi repo. https://github.com/KomodoPlatform/komodo-defi-framework/blob/14eeaeeb3da0745f7f4af1bf241e9c2d8755bc50/mm2src/common/expirable_map.rs

Can we have it as a dep, like you did with proxy sig?
```
proxy_signature = { git = "https://github.com/KomodoPlatform/komodo-defi-framework", rev = "9ebc006" }
```

Reply:
I didn't want to add the large `common` crate from KDF just to use this simple module.